application/templates/httproute.yaml

```
@@ -0,0 +1,27 @@
+{{- if and (.Values.httpRoute).enabled -}}
+{{- if not (.Capabilities.APIVersions.Has "gateway.networking.k8s.io/v1") }}
+  {{- fail "There is no APIGateway resource definition in the target cluster!" }}
+{{- end }}
+---
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: {{ template "application.name" . }}
+  namespace: {{ include "application.namespace" . }}
+  labels:
+  {{- include "application.labels" $ | nindent 4 }}
+{{- if .Values.httpRoute.additionalLabels }}
+{{ toYaml .Values.httpRoute.additionalLabels | nindent 4 }}
+{{- end }}
+{{- if .Values.httpRoute.annotations }}
+  annotations:
+{{- toYaml .Values.httpRoute.annotations | nindent 4 }}
+{{- end }}
+spec:
+  parentRefs:
+    {{- toYaml .Values.httpRoute.parentRefs | nindent 4 }}
```

I feel like fields should not be added to the template if they are not defined.

```suggestion
  {{- if .Values.httpRoute.parentRefs }}
  parentRefs:
    {{- toYaml .Values.httpRoute.parentRefs | nindent 4 }}
  {{- end }}
```

or

```suggestion
  {{- if .Values.httpRoute.parentRefs }}
  parentRefs: {{- toYaml .Values.httpRoute.parentRefs | nindent 4 }}
  {{- end }}
```